runtime/server/auth/permissions.go

```diff
@@ -0,0 +1,19 @@
+package auth
+
+// Permission represents runtime access permissions.
+type Permission int
+
+const (
+	// System-level permissions
+	ManageInstances Permission = 0x01
+
+	// Instance-level permissions
+	ReadInstance  Permission = 0x11
```

Comment:
do we tie these permissions to roles ? For example a viewer on rill should have certain set of permissions to have seamless experience, editor should have certain permissions etc.

Reply:
Probably we should note down certain user personas and list what all permissions are required for them. For example dashboard viewer on cloud will need to have all `ReadXXX` permissions ?

Reply:
Also should we plan to check resources names like metrics-view name with `ReadMetrics` to restrict access to certain dashboards etc. Probably this will come later.

Reply:
Yeah exactly, these will be mapped to some roles on the admin side. Haven't come up with them yet, but I'm picturing maybe an "admin", "contributor" and "dashboard viewer" on the admin side?

Reply:
lets create a doc for this